### DISCUSSION TITLE
Detect brand candidates from barcode prefixes

### DISCUSSION THREAD

**Discussion by CloCkWeRX** [openfoodfacts/robotoff#1206] [Ideas]
https://world.openfoodfacts.org/brand/woolworths

Observations:
 - A large majority of these items have a barcode prefix of 9300633 (a search by GTIN confirms the Woolworths Group as the owner of that prefix)
 - There is frequently common branding in the form of a logo

Suggestion:
 - Find all products grouped by the first 7 digits of a barcode having count > 10 and brand is not null
 - If 10 products all have the same branding, generate branding suggestions

Weaknesses of the approach - renaming of a brand or company could cause future products to be tagged historically.

**Comment by hangy** [comment]
FWIW, we can already find those products using the website: https://world.openfoodfacts.org/brand/-woolworths/code/9300633xxxxxx

Also, it might be worth looking into Robotoff, because it could do this based on the OCR information: openfoodfacts/robotoff#72

**Comment by CloCkWeRX** [comment]
Sample candidates for Australia; using excel:

https://world.openfoodfacts.org/cgi/search.pl?action=process&tagtype_0=states&tag_contains_0=contains&tag_0=brands-completed&tagtype_1=countries&tag_contains_1=contains&tag_1=Australia&sort_by=unique_scans_n&page_size=500&axis_x=energy&axis_y=products_n&action=display


-> LEFT(barcode, 7)
-> Group by, filter results > 10 products

https://world.openfoodfacts.org/code/9300601xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300602xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300605xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300617xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300624xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300633xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300639xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300644xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300650xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300652xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300657xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300658xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300675xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300681xxxxxx/brands.json
https://world.openfoodfacts.org/code/9300682xxxxxx/brands.json
https://world.openfoodfacts.org/code/9310012xxxxxx/brands.json
https://world.openfoodfacts.org/code/9310015xxxxxx/brands.json
https://world.openfoodfacts.org/code/9310036xxxxxx/brands.json
https://world.openfoodfacts.org/code/9310055xxxxxx/brands.json
https://world.openfoodfacts.org/code/9310072xxxxxx/brands.json
https://world.openfoodfacts.org/code/9310179xxxxxx/brands.json
https://world.openfoodfacts.org/code/9310232xxxxxx/brands.json
https://world.openfoodfacts.org/code/9310645xxxxxx/brands.json
https://world.openfoodfacts.org/code/9310653xxxxxx/brands.json
https://world.openfoodfacts.org/code/9310761xxxxxx/brands.json
https://world.openfoodfacts.org/code/9315596xxxxxx/brands.json
https://world.openfoodfacts.org/code/9319133xxxxxx/brands.json
https://world.openfoodfacts.org/code/9556041xxxxxx/brands.json

If the advanced search offered group by / count / etc filters; could be rid of the excel step.

Alternatively, on the product page/mobile UIs; could simply populate autocomplete suggestions for Brand:
```
$.get('https://world.openfoodfacts.org/code/9556041xxxxxx/brands.json', function(result) { console.log(result.tags.map(function(t) { return [t.id, t.name]; })); });
```
produces
```
0: (2) ["ayam", "Ayam"]
1: (2) ["ayam-brand", "Ayam-brand"]
2: (2) ["guinea-foods-sdn-bhd", "Guinea-foods-sdn-bhd"]
3: (2) ["duchef", "Duchef"]
4: (2) ["guinea-s", "Guinea-s"]
```

**Comment by raphael0202** [comment]
Brand prefixes are computed on Robotoff: https://github.com/openfoodfacts/robotoff/blob/master/data/brand_prefix.json. It is for the moment used as a safeguard to filter incorrect brand prediction, but we can definitely use it to predict the brand directly.
